scripts/project.js

```diff
@@ -34,7 +34,7 @@
             Project.loadAll(data);
             localStorage.setItem("etag", JSON.stringify(storedEtag));
             localStorage.setItem('rawData', JSON.stringify(data));
-            projectView.initIndexPage();
+            callback();
```

Comment:
You're calling callback() here, but you never pass a callback into the function. (and above this block you call initIndexPage() directly) This is probably causing some of your odd behavior.